### ISSUE TITLE
C++: Inherited virtual functions duplicated when `using Base::func`

### ISSUE THREAD

**Issue by ZedThree** [doxygen/doxygen#10588]
**Describe the bug**

If a derived class has a `using` declaration to bring in overloads of a virtual function, the overridden functions from the base class still appear in the output. Here's a minimal example:

```cpp
struct Base {
  /// Zero arg impl
  virtual int foo() { return 42; };
  /// One arg impl
  virtual int foo(int x) { return x; };
};

struct Derived : public Base {
  using Base::foo;
  /// One arg override
  int foo(int x) override { return x + 1; }
};
```

(source + doxyfile: [mvce.tar.gz](https://github.com/doxygen/doxygen/files/14027517/mvce.tar.gz))

which generates:

![image](https://github.com/doxygen/doxygen/assets/1486942/f2d25d5c-fbce-466b-a901-5ed4e13f2c93)

I would expect the `virtual int (int x)` function to *not* appear here, because the `override` hides it.

**Version**

This happens with 1.10.0 on openSUSE Tumbleweed 20240103, but I also compiled 1.8.20 and 1.9.0 from source and found the same issue. I couldn't easily compile older versions.

**Comment by albert-github** [comment]
In the documentation of the Base class I see also something peculiar:

![image](https://github.com/doxygen/doxygen/assets/5223533/c7eb2fcd-a730-4c4c-8261-b812537d9d08)

I.e. twice the that the function is reimplemented in `Derived`

It looks like that:
```
  using Base::foo;
```
is causing the problems, when I comment this line the result looks like what OP expects.

**Comment by ZedThree** [comment]
Yes, sorry it is exactly the [`using` declaration](https://en.cppreference.com/w/cpp/language/using_declaration) that causes the issue. Without this line the `Base::foo()` (zero-arg) overload is hidden by the `Derived::foo(int)` method, so this is required for valid code.

Removing that line generates:

![image](https://github.com/doxygen/doxygen/assets/1486942/2badec2d-12d0-4e1b-86ee-aa37c4defc08)

which is what I expect.

**Comment by doxygen** [comment]
@ZedThree Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by ZedThree** [comment]
@doxygen Thanks, this fixes it for me!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
